x-pack/plugins/endpoint/server/routes/resolver/utils/pagination.ts

```diff
@@ -0,0 +1,91 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { SearchResponse } from 'elasticsearch';
+import { ResolverEvent } from '../../../../common/types';
+import { extractEventID } from './normalize';
+import { JsonObject } from '../../../../../../../src/plugins/kibana_utils/public';
+
+export interface PaginationParams {
+  size: number;
+  timestamp?: Date;
+  eventID?: string;
+}
+
+interface PaginationCursor {
+  timestamp: Date;
+  eventID: string;
+}
+
+function urlEncodeCursor(data: PaginationCursor) {
+  const value = JSON.stringify(data);
```

Comment:
❔ JSON operations are frequent sources of errors: Should there be a defensive try/catch around this?

Reply:
Thanks for point that out, I added it to this issue: https://github.com/elastic/endpoint-app-team/issues/180 that outlines the work to clean up some of the error handling in the implementation.

Reply:
I don't think we'll have to worry about that in this particular case--even if it errors, we'll bubble up and just 500 from the handler itself--but, in this case, what we're doing is literally just taking some well-defined previously JSON-encoded fields and re-encoding them--so something would have to be pretty broken to error in this case.